block/manager.go

```diff
@@ -125,7 +125,16 @@ func (m *Manager) SetDALC(dalc da.DataAvailabilityLayerClient) {
 }
 
 func (m *Manager) AggregationLoop(ctx context.Context) {
+	genesisTime := m.genesis.GenesisTime
+	height := m.store.Height()
+	if height == 0 && time.Until(genesisTime) > 0 {
+		time.Sleep(time.Until(genesisTime))
+	} else if height > 0 && time.Until(m.lastState.LastBlockTime.Add(m.conf.BlockTime)) > 0 {
```

Comment:
`time.Until(m.lastState.LastBlockTime.Add(m.conf.BlockTime))` could be extracted into variable.

Actually, consider using something like:
```
var delay time.Duration
if height == m.genesis.InitialHeight {
    delay = ...
} else {
    delay = ...
}
if delay > 0 {
    log
    sleep
}
```

Reply:
This looks good, I will refactor it into this